File: leetCode/math/CanMakeArithmeticProgressionFromSequence.py
# https://leetcode.com/problems/can-make-arithmetic-progression-from-sequence/
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        min_a = min(arr)
        max_a = max(arr)
        set_a = set(arr)
        r = (max_a - min_a) // (len(arr) - 1)
        
        for a in arr:
            if min_a not in set_a:
                return False
            min_a += r
        
        return True

File: leetCode/array/can-make-arithmetic-progression-from-sequence.rs
// https://leetcode.com/problems/can-make-arithmetic-progression-from-sequence/
impl Solution {
    pub fn can_make_arithmetic_progression(arr: Vec<i32>) -> bool {
        if arr.len() < 2 {
            return false;
        }
        
        let mut arr = arr;
        arr.sort();
        let diff = arr[1] - arr[0];

        for i in 2..arr.len() {
            if arr[i] - arr[i - 1] != diff {
                return false;
            }
        }

        true
    }
}
